rpsl-tools/main.go

```diff
@@ -0,0 +1,186 @@
+//
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// irrParser loads the content of IRR database backups, parses the content
+// with the rpsl-parser library and generates loadable data for indexed
+// queries of the irr content.
+package main
+
+import (
+	"compress/gzip"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	glog "github.com/golang/glog"
+	rpsl "github.com/manrs-tools/contrib/rpsl-parser"
+)
+
+var (
+	threads = flag.Int("threads", 4, "Max threads to use in parsing db files.")
+)
+
+// Define a slice of filenames to get as a single flag on startup.
+type fileSlice []string
+
+var rpslFiles fileSlice
+
+// Stirng and Set are required for the variadic flag: rpslFiles.
+func (f *fileSlice) String() string {
+	return fmt.Sprintf("%v", *f)
+}
+
+func (f *fileSlice) Set(value string) error {
+	for _, fn := range strings.Split(value, ",") {
+		*f = append(*f, fn)
+	}
+	return nil
+}
+
+// getReader parses the filename and either returns a gzip reader or regular reader.
+func getReader(fn string) (io.Reader, error) {
+	fd, err := os.Open(fn)
+	if err != nil {
+		fmt.Printf("Failed to open file(%v): %v\n", fn, err)
+		return nil, err
+	}
+	if strings.HasSuffix(fn, ".gz") {
+		f, err := gzip.NewReader(fd)
+		if err != nil {
+			fmt.Printf("Failed to read gzip'd file(%v): %v\n", fn, err)
+			return nil, err
+		}
+		return f, nil
+	}
+	return fd, nil
+}
+
+// parseFile parses a single file at a time into records sent to the
+// records channel (rc). Files arrive as names on the input channel (ic).
+func parseFile(ic <-chan string, rc chan<- *rpsl.Record, ec chan<- bool) {
+	for fn := range ic {
+		var rdr *rpsl.Reader
+
+		reader, err := getReader(fn)
+		if err != nil {
+			glog.Infof("failed to read file(%v): %v\n", fn, err)
```

Comment:
nit:  s/failed/Failed to be consistent with other err log messages.

Reply:
done, thanks!